openmdao/core/component.py

```diff
@@ -273,6 +268,18 @@ def _setup_var_data(self, recurse=True):
                 # Compute abs2meta
                 abs2meta[abs_name] = metadata
 
+            for prom_name, val in iteritems(self._var_discrete[type_]):
+                abs_name = prefix + prom_name
+                allprocs_prom2abs_list[type_][prom_name] = [abs_name]
+                self._var_allprocs_discrete[type_][abs_name] = val
+
+        self._var_abs_names = self._var_allprocs_abs_names
+        if self._var_discrete['input'] or self._var_discrete['output']:
+            self._discrete_inputs = _DictValues(self._var_discrete['input'])
```

Comment:
Is there a docstring for _discrete_inputs and outputs somewhere?

Reply:
fixed